File: Lancha.py
from embarcacion import Embarcacion

class Lancha(Embarcacion):
    def __init__(self, nombre, capacidad, longitud, potencia_motor):
        super().__init__(nombre, "Lancha", capacidad, longitud)
        self.potencia_motor = potencia_motor

    def mostrar_info(self):
        super().mostrar_info()
        print(f"Potencia del motor: {self.potencia_motor} HP")

    def acelerar(self):
        print(f"La lancha {self.nombre} está acelerando con {self.potencia_motor} HP.")

File: Lancha.cpp
#include "Lancha.h"

Lancha::Lancha(string n, int c, double l, int a, int p, double comb)
    : Embarcacion(n, c, l, a) {
    potenciaMotor = p;
    combustible = comb;
    cout << "Lancha creada: " << nombre << endl;
}

Lancha::~Lancha() {
    cout << "Lancha destruida: " << nombre << endl;
}

void Lancha::acelerar() {
    if (combustible > 0) {
        cout << nombre << " está acelerando..." << endl;
        combustible -= 5;
    } else {
        cout << nombre << " no tiene combustible." << endl;
    }
}

void Lancha::mostrarDatos() {
    cout << "\n=== DATOS DE LA LANCHA ===" << endl;
    Embarcacion::mostrarDatos();
    cout << "Potencia del motor: " << potenciaMotor << " HP" << endl;
    cout << "Combustible: " << combustible << " litros" << endl;
}
